src/Misc/WaveShapeSmps.cpp

```diff
@@ -79,16 +123,18 @@ void waveShapeSmps(int n,
             break;
         case 7:
             ws = powf(2.0f, -ws * ws * 8.0f); //Limiter
+            par = par/4;
+            if (par > ws - 0.01) par = ws - 0.01;
             for(i = 0; i < n; ++i) {
-                float tmp = smps[i];
-                if(fabs(tmp) > ws) {
-                    if(tmp >= 0.0f)
-                        smps[i] = 1.0f;
-                    else
-                        smps[i] = -1.0f;
-                }
+                smps[i] += offs;
+                float res = polyblampres(smps[i], ws, par);
+
+                if (smps[i]>=0)
+                    smps[i] = ( smps[i] > ws ? ws-res : smps[i]-res );
                 else
-                    smps[i] /= ws;
+                    smps[i] = ( smps[i] < -ws ? -ws+res : smps[i]+res );
```

Comment:
Previously, the increase in the non clipped region was dependent on `ws`, but now, it's always 1 (plus the minor `res`). Did you forget to divide by `ws`?

Reply:
you are right. i will fix it.

Reply:
should be in the patch in https://sourceforge.net/p/zynaddsubfx/patches/43/#f252